core/commands/storage.go

```diff
@@ -153,9 +159,33 @@ Receive proofs as collateral evidence after selected nodes agree to store the fi
 			shardHashes []string
 			rootHash    cidlib.Cid
 			shardSize   uint64
+			blacklist   = set.New()
+			renterId    = n.Identity
 		)
-		lf, _ := req.Options[leafHashOptionName].(bool)
-		if lf == false {
+		isRepairMode := req.Options[repairModeOptionName].(bool)
+		lf := req.Options[leafHashOptionName].(bool)
+		if isRepairMode {
+			if len(req.Arguments) > 3 {
+				blacklistStr := req.Arguments[3]
+				for _, s := range strings.Split(blacklistStr, ",") {
+					blacklist.Add(s)
+				}
+			}
+			rootHash, err = cidlib.Parse(req.Arguments[0])
+			renterId, err = peer.IDB58Decode(req.Arguments[2])
+			if err != nil {
+				return err
+			}
+			shardHashes = strings.Split(req.Arguments[1], ",")
+			shardCid, err := cidlib.Parse(shardHashes[0])
```

Comment:
should be fine now, but I am wondering why not check all shardHash passed from guard?

Reply:
i just copy the codes. but i think it's because here we just want to get the first shardcid and parse the shard size from it.